src/main/java/org/jboss/ejb/protocol/remote/EJBClientChannel.java

```diff
@@ -368,6 +383,9 @@ private void processMessage(final MessageInputStream message) {
                                 continue;
                             }
                             final ServiceRegistration abstractRegistration = persistentClusterRegistry.registerService(abstractBuilder.create());
+                            if (Logs.INVOCATION.isDebugEnabled()) {
+                                Logs.INVOCATION.debugf("Received CLUSTER_TOPOLOGY(%x) message, registering ServiceURL(%s)", msg, abstractBuilder.create());
```

Comment:
Better to cache abstractBuilder.create() into a local variable.  Then you don't need isXxxEnabled().